### DISCUSSION TITLE
Are Filament layouts (specifically the main navigation) also rendered by Livewire?

### DISCUSSION THREAD

**Discussion by realslimsutton** [filamentphp/filament#3672] [General]
I'm attempting to dynamically add items to the navigation after clicking an action. This works great so far, but requires me to refresh the page for the items to appear in the navigation.

I've tried emitting two events to test if I can force the navigation to be updated after clicking an action, but neither of which seem to make any difference (which leads me to believe that the navigation isn't apart of Livewire or at least I'm unable to force it to be updated). This is inside a Resource (specifically the ListRecords page):

**ListModules.php**
```
use Filament\Resources\Pages\ListRecords;

class ListModules extends ListRecords
{
    protected $listeners = [
        'modulesUpdated' => '$refresh
        'modulesUpdated2' => 'render'
    ];

   public function enableModule()
   {
        // Enable the module here

        $this->emit('modulesUpdated');
        $this->emit('modulesUpdated2');
    }
}
```

**Comment by danharrin** [comment]
They are inside the layout, which is not managed by Livewire. This reduces the amount of HTML we return from each request.

You could redirect to the current page after the action is submitted to refresh them?

**Reply by realslimsutton** [reply]
Yep that makes sense, and is also a sensible solution, thanks!
